lib/msal-browser/README.md

```diff
@@ -38,14 +72,129 @@ We plan to continue support for the implicit flow in the library.
+
+It is best practice to only request scopes you need when you need them, a concept called dynamic consent. While this can create more interactive consent for users in your application, it also reduces drop-off from users that may be uneasy granting a large list of permissions for features they are not yet using.
+
+AAD will only allow you to get consent for 3 resources at a time, although you can request many scopes within a resource.
+When the user makes a login request, you can pass in multiple resources and their corresponding scopes because AAD issues an idToken pre consenting those scopes. However acquireToken calls are valid only for one resource / multiple scopes. If you need to access multiple resources, please make separate acquireToken calls per resource.
+
+```JavaScript
+var loginRequest = {
+    scopes: ["user.read", "mail.send"] // optional Array<string>
+};
+
+msalInstance.loginPopup(loginRequest)
+.then(response => {
+    // handle response
+})
+.catch(err => {
+    // handle error
+});
+```
+
+#### 4. Get an access token to call an API
+
+In MSAL, you can get access tokens for the APIs your app needs to call using the `acquireTokenSilent` method which makes a silent request(without prompting the user with UI) to Azure AD to obtain an access token. The Azure AD service then returns an [access token](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) containing the user consented scopes to allow your app to securely call the API.
+
+You can use `acquireTokenRedirect` or `acquireTokenPopup` to initiate interactive requests, although, it is best practice to only show interactive experiences if you are unable to obtain a token silently due to interaction required errors. If you are using an interactive token call, it must match the login method used in your application. (`loginPopup`=> `acquireTokenPopup`, `loginRedirect` => `acquireTokenRedirect`).
+
+If the `acquireTokenSilent` call fails, you may need to initiate an interactive request. This could happen for many reasons including scopes that have been revoked, expired tokens, or password changes.
+
+`acquireTokenSilent` will look for a valid token in the cache, and if it is close to expiring or does not exist, will automatically try to refresh it for you.
+
+See [Request and Response Data Types]() for reference.
+
+```JavaScript
+    // if the user is already logged in you can acquire a token
+    if (msalInstance.getAccount()) {
+        var tokenRequest = {
+            scopes: ["user.read", "mail.send"]
+        };
+        msalInstance.acquireTokenSilent(tokenRequest)
+            .then(response => {
+                // get access token from response
+                // response.accessToken
+            })
+            .catch(err => {
+                // handle error
+            });
+    } else {
+        // user is not logged in, you will need to log them in to acquire a token
```

Comment:
Could have a sample that calls one of the login methods.

Reply:
We don't throw interaction required anymore, need to do more research to understand how this will work.

Reply:
Ah, I'm guessing this will be status code based now.